plugin/src/main/kotlin/com/vanniktech/maven/publish/sonatype/SonatypeRepositoryBuildService.kt

```diff
@@ -42,24 +43,26 @@ internal abstract class SonatypeRepositoryBuildService : BuildService<SonatypeRe
   // indicates whether we already closed a staging repository to avoid doing it more than once in a build
   var repositoryClosed: Boolean = false
 
-  var buildHasFailure: Boolean = false
+  private var buildIsSuccess: Boolean = true
 
   override fun onFinish(event: FinishEvent) {
-    if (event.result is TaskFailureResult) {
-      buildHasFailure = true
+    if (event.result is FailureResult) {
+      buildIsSuccess = false
     }
   }
 
   override fun close() {
-    if (buildHasFailure) {
-      return
-    }
-
     val stagingRepositoryId = this.stagingRepositoryId
     if (stagingRepositoryId != null) {
-      nexus.closeStagingRepository(stagingRepositoryId)
-      if (parameters.automaticRelease.get()) {
-        nexus.releaseStagingRepository(stagingRepositoryId)
+      if (buildIsSuccess) {
+        nexus.closeStagingRepository(stagingRepositoryId)
+        if (parameters.automaticRelease.get()) {
+          nexus.releaseStagingRepository(stagingRepositoryId)
+        }
+      } else {
+        try {
+          nexus.dropStagingRepository(stagingRepositoryId)
+        } catch (_: IOException) {}
```

Comment:
Should we log it?

Reply:
When the build fails people often jump to the last message that looks like an error which in this case would be misleading because this didn't fail the build. Maybe as an info level log? Then you could find out about it when running with `--info` but by default Gradle doesn't print it

Reply:
Sounds good!